notebooks/04-esmvaltool-singularity/04-esmvaltool-singularity.ipynb

```diff
@@ -0,0 +1,285 @@
+   "id": "1153b184-5877-4d7c-889a-15311698f33b",
+   "metadata": {
+    "jp-MarkdownHeadingCollapsed": true,
+    "tags": []
+   },
+   "source": [
+    "-- Please click this block, and add a Dask cluster using the Dask tab on the left of your Jupyter terminal. --\n",
+    "\n",
+    "-- You can first click the `Scale` button to set up the number of workers. Then click `<>` to add a code block --\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 20,
+   "id": "4742f29d-1432-4221-ac47-e4c5316a1b40",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "['singularity run /home/caroline-oku/demo_singularity/demo_singularity_esmvaltool/esmvaltool_stable.sif run /home/caroline-oku/demo_singularity/demo_singularity_esmvaltool/recipes/recipe_dataset1.yml --offline=False',\n",
+       " 'singularity run /home/caroline-oku/demo_singularity/demo_singularity_esmvaltool/esmvaltool_stable.sif run /home/caroline-oku/demo_singularity/demo_singularity_esmvaltool/recipes/recipe_dataset2.yml --offline=False']"
+      ]
+     },
+     "execution_count": 20,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Set up the commands for execution\n",
+    "import os\n",
+    "\n",
+    "# Get the abusolte path for the working directory\n",
+    "cwd = os.getcwd()\n",
+    "f_sif = '{}/esmvaltool_stable.sif'.format(cwd)\n",
+    "dir_recipe = '{}/recipes'.format(cwd)\n",
+    "\n",
+    "# Two recipes for two datasets\n",
+    "recipes = ['recipe_dataset1.yml', 'recipe_dataset2.yml']\n",
+    "\n",
+    "# Set up shells commands\n",
+    "commands = ['singularity run {} run {}/{} --offline=False'.format(f_sif, dir_recipe, r) for r in recipes]\n",
+    "commands"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4ccf37b8-4f4b-4279-bc68-b03d8007f3c2",
+   "metadata": {},
+   "source": [
+    "One can submit the commands to the Dask clusteras follow:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "64aa907b-2c4e-4f2b-9bd3-e31936e87508",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Submit the commands\n",
+    "future = client.map(os.system, commands)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "id": "9f85f72c-2c83-477c-8784-76c2b10dd788",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "[<Future: finished, type: int, key: system-e879bb29e07bc8513966c1f2d5a62896>,\n",
+       " <Future: finished, type: int, key: system-35008dcc096c451ca0fbfe5cb37433a3>]"
+      ]
+     },
+     "execution_count": 15,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "future"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9393720e-bc0c-4722-b2fb-14a82ac1ba9b",
+   "metadata": {},
+   "source": [
+    "Once finished, one can check the downloaded climate data files and the generated results:"
```

Comment:
This is great. I would be very curios to know if/to what extent it is feasible to use ESMvaltool to also generate intermediate data products, like the input for the time series for example. Store them, and then retrieve and run a different recipe with them to generate final plots, for example. Possibly for the future